src/main.rs

```diff
@@ -1,26 +1,38 @@
 mod block_tracer;
 mod cache;
+mod cli;
 mod general_trace_comparison;
 mod juno_manager;
 mod transaction_simulator;
 mod transaction_tracer;
 
+use crate::cli::{Cli, Commands};
 use block_tracer::{BlockTracer, TraceBlockReport};
 use cache::get_sorted_blocks_with_tx_count;
 use chrono::Local;
-use clap::{arg, command, value_parser, ArgAction, Command};
-use core::panic;
-use env_logger::Env;
+use clap::Parser;
 use general_trace_comparison::generate_block_comparison;
 use juno_manager::{JunoBranch, JunoManager, ManagerError};
-use log::{error, info, warn};
+use log::{info, warn};
 use std::io::Write;
 use std::path::Path;
 use tokio::fs::OpenOptions;
 use tokio::io::{AsyncWriteExt, BufWriter};
 use transaction_simulator::{log_block_report, SimulationStrategy, TransactionSimulator};
 use transaction_tracer::TraceResult;
 
+const RERUNNING_AFTER_FIXES: bool = true;
+
+fn should_run_block(block_number: &u64) -> bool {
+    RERUNNING_AFTER_FIXES
+        || (!Path::new(&format!("./results/{block_number}.json")).exists()
+            && !Path::new(&format!("./results/trace-{block_number}.json")).exists()
+            && ![
+                610508_u64, 610541_u64, 612572_u64, 612787_u64, 613138_u64, 613978_u64,
+            ]
+            .contains(block_number))
```

Comment:
I think this is old code right? Why are you adding this back in? Why there is a fix set of blocks in the code?
What happened with the cache mechanism written by Pearson?

Also, you are missing the flag for the cache mechanism as well I think

Reply:
It seems like some conflicts with local versioning were around, I will fix that 👍